File: spojPractice/spoj1724.java
import java.io.*;
import java.math.*;
class spoj1724
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int t=Integer.parseInt(br.readLine());
		while(t--!=0)
		{
			int n=Integer.parseInt(br.readLine());
			BigInteger sum=BigInteger.ZERO;
			sum=BigInteger.valueOf((long)Math.pow(n,2));
			
			n--;
			if(n%2==0)
			{
				int x=n/2;
				sum=sum.add(BigInteger.valueOf(4*(x)*(x+1)*(2*x+1)/6));
			}
			else
			{
				int x=(n-1)/2+1;
				sum=sum.add(BigInteger.valueOf(4*(x)*(x+1)*(2*x+1)/6+x-2*(x)*(x+1)));
			}
			System.out.println(sum);
		}
	}
}

File: spojPractice/spoj1724.cpp
#include <iostream>
#include <math.h>
using namespace std;
int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		long long n;
		cin>>n;
		long long sum=0;
		if(n%2==1)
		{
			sum=pow(n,2)+pow(n-1,2);
			long long p=(n-3)/2;
			sum+=(2*(p)*(p+1)*(2*p+1))/3;
			cout<<sum<<endl;
		}
		else
		{
			sum=pow(n,2)+pow(n-1,2);
			long long p=(n-2)/2;
			sum+=(2*(p)*(p+1)*(2*p+1))/3+p-2*(p)*(p+1);
			cout<<sum<<endl;
		}
	}
}